packages/utils/src/conversions.ts

```diff
@@ -176,9 +177,79 @@ export function bigNumberishToBigInt(n: BigNumberish): bigint {
  * @returns The buffer representation of the BigNumberish value.
  */
 export function bigNumberishToBuffer(n: BigNumberish): Buffer {
-    if (n instanceof Buffer) {
-        return n
+    if (isBuffer(n)) {
+        return n as Buffer
     }
 
     return bigIntToBuffer(bigNumberishToBigInt(n))
 }
+
+/**
+ * Converts an hexadecimal string to a buffer. The hexadecimal string
+ * should not start with '0x' or '0X'.
+ * It uses a big-endian byte order.
```

Comment:
I'm not sure that endian-ness is meaningful on byte order kind of conversion.  There's no "most significant byte" in either the input or the output, just a sequence of bytes indexed 0...N.  Byte order comes up when converting bytes into numbers which are bigger than a byte.

I'd suggest you only have one version of this conversion (the one without the reverse) and simply explain that it keeps the bytes in the same order, after grouping the hex digits in 2-digit chunks.

Technically (but not relevant and not worth changing or documenting) the conversion of 2 hex digits to a byte has a digit order (not a byte order) which is big-endian, but as with normal hex and decimal, I think that's obvious enough to be assumed.  Digits always put the most-significant digit first.  You'd never write 5*2 as 01 rather than 10.

Reply:
Oh this byte-order stuff still confuses me a bit 😅  Thanks for this explanation!

Reply:
It's super unclear, and I probably don't have the terminology perfect either.  Thanks for putting up with my pedantry.

Reply:
Oh no, please be pedantic. It's what a PR needs 👍🏽  Thanks!